src/main/kotlin/no/skatteetaten/aurora/boober/feature/HerkimerVaultFeature.kt

```diff
@@ -0,0 +1,86 @@
+package no.skatteetaten.aurora.boober.feature
+
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.fkorotkov.kubernetes.metadata
+import com.fkorotkov.kubernetes.newSecret
+import io.fabric8.kubernetes.api.model.Secret
+import no.skatteetaten.aurora.boober.model.AuroraConfigFieldHandler
+import no.skatteetaten.aurora.boober.model.AuroraContextCommand
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.model.AuroraResource
+import no.skatteetaten.aurora.boober.model.addEnvVarsToMainContainers
+import no.skatteetaten.aurora.boober.service.HerkimerService
+import no.skatteetaten.aurora.boober.service.ResourceKind
+import no.skatteetaten.aurora.boober.utils.boolean
+import no.skatteetaten.aurora.boober.utils.createEnvVarRefs
+import no.skatteetaten.aurora.boober.utils.findResourcesByType
+import no.skatteetaten.aurora.boober.utils.jsonMapper
+import org.apache.commons.codec.binary.Base64
+import org.springframework.stereotype.Service
+
+private const val FEATURE_FIELD = "vault"
+
+@Service
+class HerkimerVaultFeature(
+    val herkimerService: HerkimerService
+) : Feature {
+    override fun handlers(header: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraConfigFieldHandler> {
+        return setOf(
+            AuroraConfigFieldHandler("$FEATURE_FIELD/enabled", validator = { it.boolean() }),
+            AuroraConfigFieldHandler("$FEATURE_FIELD/prefix", defaultValue = "")
+
+        )
+    }
+
+    override fun validate(
+        adc: AuroraDeploymentSpec,
+        fullValidation: Boolean,
+        cmd: AuroraContextCommand
+    ): List<Exception> {
+        return emptyList()
+    }
+
+    override fun generate(adc: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraResource> {
+        if (!adc.isFeatureEnabled()) return emptySet()
+
+        val nameAndCredentials =
+            herkimerService.getClaimedResources(adc.applicationDeploymentId, ResourceKind.ManagedPostgresDatabase)
+                .associate { resource ->
+                    resource.name to resource.claims.firstOrNull()?.credentials
```

Comment:
hva bør navnet på ressursen i herkimer være for DatabaseInstance? InstanceName kan vel bli litt feil med tanke på at man vil få et management cluster etterhvert. Hva med hostnavnet?